airflow/www/views.py

```diff
@@ -1804,53 +1802,86 @@ def _mark_task_instance_state(  # pylint: disable=too-many-arguments
         latest_execution_date = dag.get_latest_execution_date()
         if not latest_execution_date:
             flash(f"Cannot make {state}, seem that dag {dag_id} has never run", "error")
-            return redirect(origin)
+            return redirect(request.referrer)
```

Comment:
Referrer is not always sent (some AV blocks it for instance) so we can't just do this.

I also wonder if this makes it function as an open redirector of sorts

Reply:
@ashb this is inside the POST call, so the only way to hit this endpoint via the confirm page.. so in that case, the referrer should be always available?

Reply:
Nope, referrer can still be blocked

Reply:
Ok, brought the `origin` back. In case someone is clicking it via an email link, then after the operation, it loads the home page, which is fine in my opinion.